packages/extension/src/contexts/NetworkContext/NetworkContext.tsx

```diff
@@ -55,19 +55,25 @@ const NetworkProvider: FC = ({ children }) => {
   useEffect(() => {
     let retryCount = 0;
     const maxRetries = 3;
+    let nodeIndex = 0;
 
     const connectToNetwork = async () => {
       const network = loadNetwork();
       setNetworkName(network.name);
+      let node = network.server;
+      if (nodeIndex !== 0 && network.nodes?.[nodeIndex]) {
+        node = network.nodes[nodeIndex];
+      }
```

Comment:
else {
// either throw an exception or reset the index to 0 because it means that we have tried all the servers from the list and none worked 
}